modules/codegen/src/smithy4s/codegen/transformers/AwsStandardTypesTransformer.scala

```diff
@@ -0,0 +1,99 @@
+package smithy4s.codegen.transformers
+
+import smithy4s.codegen.transformers.AwsStandardTypesTransformer.MemberShapeBuilderOps
+import software.amazon.smithy.build.{ProjectionTransformer, TransformContext}
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes._
+import software.amazon.smithy.model.traits.{BoxTrait, DefaultTrait, Trait}
+
+class AwsStandardTypesTransformer extends ProjectionTransformer {
+
+  private val smithyStandardNamespace = "smithy.api"
+  private val awsNamespacePrefix = "com.amazonaws"
+
+  override def getName: String = AwsStandardTypesTransformer.name
+
+  override def transform(context: TransformContext): Model = {
+    val transformer = context.getTransformer
+
+    transformer.removeShapesIf(
+      transformer.mapShapes(context.getModel, mapFunction(context.getModel)),
+      canReplace
+    )
+  }
+
+  private def mapFunction(model: Model): java.util.function.Function[Shape, Shape] = shape => {
+    def replaceWith(shapeId: ShapeId): ShapeId = {
+      val shapeType =
+        model
+          .expectShape(shapeId) // this should be safe as we already did some filtering before calling the method
+          .getType
+
+      // Use the shape type instead of relying on the name
+      // because we want to be able to replace something
+      // com.amazonaws.dynamodb#Date with smithy.api#Timestamp
+      val shapeName = shapeType.getShapeClass.getSimpleName.replace("Shape", "")
+
+      ShapeId.fromParts(smithyStandardNamespace, shapeName)
+    }
+
+    shape.asMemberShape()
+      .filter{ memberShape =>
+        model.getShape(memberShape.getTarget)
+          .map[Boolean](canReplace)
+          .orElse(false)
+      }
+      .flatMap[Shape](shape => {
+        val replacementShape = replaceWith(shape.getTarget)
+        val target = model.expectShape(shape.getTarget)
+
+        val builder = MemberShape
+          .builder()
+          .id(shape.getId)
+          .source(shape.getSourceLocation)
+          .target(replacementShape)
+          .copyDefaultTrait(target, replacementShape)
```

Comment:
this is flawed as it doesn't keep the traits of the original member. Also there's a simpler way to achieve what you're trying to do : 

```patch
diff --git a/modules/codegen/src/smithy4s/codegen/transformers/AwsStandardTypesTransformer.scala b/modules/codegen/src/smithy4s/codegen/transformers/AwsStandardTypesTransformer.scala
index 3536db21..7b41dda0 100644
--- a/modules/codegen/src/smithy4s/codegen/transformers/AwsStandardTypesTransformer.scala
+++ b/modules/codegen/src/smithy4s/codegen/transformers/AwsStandardTypesTransformer.scala
@@ -22,11 +22,15 @@ class AwsStandardTypesTransformer extends ProjectionTransformer {
     )
   }
 
-  private def mapFunction(model: Model): java.util.function.Function[Shape, Shape] = shape => {
+  private def mapFunction(
+      model: Model
+  ): java.util.function.Function[Shape, Shape] = shape => {
     def replaceWith(shapeId: ShapeId): ShapeId = {
       val shapeType =
         model
-          .expectShape(shapeId) // this should be safe as we already did some filtering before calling the method
+          .expectShape(
+            shapeId
+          ) // this should be safe as we already did some filtering before calling the method
           .getType
 
       // Use the shape type instead of relying on the name
@@ -37,25 +41,22 @@ class AwsStandardTypesTransformer extends ProjectionTransformer {
       ShapeId.fromParts(smithyStandardNamespace, shapeName)
     }
 
-    shape.asMemberShape()
-      .filter{ memberShape =>
-        model.getShape(memberShape.getTarget)
+    shape
+      .asMemberShape()
+      .filter { memberShape =>
+        model
+          .getShape(memberShape.getTarget)
           .map[Boolean](canReplace)
           .orElse(false)
       }
-      .flatMap[Shape](shape => {
-        val replacementShape = replaceWith(shape.getTarget)
+      .map[Shape] { shape =>
         val target = model.expectShape(shape.getTarget)
-
-        val builder = MemberShape
-          .builder()
-          .id(shape.getId)
-          .source(shape.getSourceLocation)
+        val replacementShape = replaceWith(shape.getTarget)
+        shape.toBuilder
           .target(replacementShape)
-          .copyDefaultTrait(target, replacementShape)
-
-        java.util.Optional.of(builder.build().asInstanceOf[Shape])
-      })
+          .copyDefaultTrait(target)
+          .build()
+      }
       .orElse(shape)
   }
 
@@ -64,14 +65,18 @@ class AwsStandardTypesTransformer extends ProjectionTransformer {
 
   private def canReplace(shape: Shape): Boolean = {
     shape.isInstanceOf[SimpleShape] &&
-      isAwsShape(shape) &&
-      onlySupportedTraits(shape.getAllTraits)
+    isAwsShape(shape) &&
+    onlySupportedTraits(shape.getAllTraits)
   }
 
   @annotation.nowarn
-  private def onlySupportedTraits(traits: java.util.Map[ShapeId, Trait]) = traits.values().stream().allMatch(t => {
-      t.isInstanceOf[DefaultTrait] || t.isInstanceOf[BoxTrait]
-  })
+  private def onlySupportedTraits(traits: java.util.Map[ShapeId, Trait]) =
+    traits
+      .values()
+      .stream()
+      .allMatch(t => {
+        t.isInstanceOf[DefaultTrait] || t.isInstanceOf[BoxTrait]
+      })
 
 }
 
@@ -79,13 +84,16 @@ object AwsStandardTypesTransformer {
 
   val name: String = "AwsStandardTypesTransformer"
 
-  private[transformers] final implicit class MemberShapeBuilderOps(val builder: MemberShape.Builder) extends AnyVal {
-    def copyDefaultTrait(shape: Shape, replacementShape: ShapeId): MemberShape.Builder = {
+  private[transformers] final implicit class MemberShapeBuilderOps(
+      val builder: MemberShape.Builder
+  ) extends AnyVal {
+    def copyDefaultTrait(
+        shape: Shape
+    ): MemberShape.Builder = {
       val defaultTraitOpt = toOption(shape.getTrait(classOf[DefaultTrait]))
 
-      defaultTraitOpt.map { df =>
-        new DefaultTrait.Provider().createTrait(replacementShape, df.toNode)
-      }.fold(builder){defaultTrait => builder.addTrait(defaultTrait)}
+      defaultTraitOpt
+        .fold(builder)(builder.addTrait(_))
     }
   }
 
@@ -96,4 +104,4 @@ object AwsStandardTypesTransformer {
       None
     }
   }
-}
\ No newline at end of file
+}
```

Reply:
Fixed + added a test to validate that member traits are kept.